packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -646,59 +675,64 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
    *
-   * @param params - The parameters for fetching the password.
+   * @param params - The parameters for unlocking the controller.
    * @param params.globalPassword - The latest global password.
-   * @returns A promise that resolves to the password corresponding to the current authPubKey in state.
+   * @returns A promise that resolves to the success of the operation.
    */
-  async recoverCurrentDevicePassword({
+  async submitGlobalPassword({
     globalPassword,
   }: {
     globalPassword: string;
-  }): Promise<{ password: string }> {
+  }): Promise<void> {
     return await this.#withControllerLock(async () => {
       return await this.#executeWithTokenRefresh(async () => {
         const currentDeviceAuthPubKey = this.#recoverAuthPubKey();
-        const { password: currentDevicePassword } = await this.#recoverPassword(
-          {
-            targetPwPubKey: currentDeviceAuthPubKey,
-            globalPassword,
-          },
-        );
-        return {
-          password: currentDevicePassword,
-        };
-      }, 'recoverCurrentDevicePassword');
+        await this.#submitGlobalPassword({
+          targetAuthPubKey: currentDeviceAuthPubKey,
+          globalPassword,
+        });
+      }, 'submitGlobalPassword');
     });
   }
 
   /**
-   * @description Fetch the password corresponding to the targetPwPubKey.
+   * @description Submit the global password to the controller, verify the
+   * password validity and unlock the controller.
    *
-   * @param params - The parameters for fetching the password.
-   * @param params.targetPwPubKey - The target public key of the password to recover.
+   * @param params - The parameters for submitting the global password.
+   * @param params.targetAuthPubKey - The target public key of the keyring
+   * encryption key to recover.
    * @param params.globalPassword - The latest global password.
-   * @returns A promise that resolves to the password corresponding to the current authPubKey in state.
+   * @returns A promise that resolves to the keyring encryption key
+   * corresponding to the current authPubKey in state.
    */
-  async #recoverPassword({
-    targetPwPubKey,
+  async #submitGlobalPassword({
+    targetAuthPubKey,
     globalPassword,
   }: {
-    targetPwPubKey: SEC1EncodedPublicKey;
+    targetAuthPubKey: SEC1EncodedPublicKey;
     globalPassword: string;
-  }): Promise<{ password: string }> {
-    const { encKey: latestPwEncKey, authKeyPair: latestPwAuthKeyPair } =
+  }): Promise<void> {
+    const { encKey: latestEncKey, authKeyPair: latestAuthKeyPair } =
       await this.#recoverEncKey(globalPassword);
 
     try {
-      const res = await this.toprfClient.recoverPassword({
-        targetPwPubKey,
-        curEncKey: latestPwEncKey,
-        curAuthKeyPair: latestPwAuthKeyPair,
+      // Recover vault encryption key.
+      const res = await this.toprfClient.recoverPwEncKey({
+        targetAuthPubKey,
+        curPwEncKey: latestEncKey,
```

Comment:
this should be latestPwdEncKe, we encrypt pwd encryption key using pwdEncKey in backup store and rest of the items using latestEncKey right?